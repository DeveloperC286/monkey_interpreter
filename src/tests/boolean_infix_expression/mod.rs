use super::*;

#[template]
#[rstest(
    code,
    snapshot_name,
    case("true == true", "boolean_infix_expression_case_1"),
    case("false == false", "boolean_infix_expression_case_2"),
    case("true == false", "boolean_infix_expression_case_3"),
    case("false == true", "boolean_infix_expression_case_4"),
    case("true != false", "boolean_infix_expression_case_5"),
    case("false != true", "boolean_infix_expression_case_6"),
    case("true != true", "boolean_infix_expression_case_7"),
    case("false != false", "boolean_infix_expression_case_8")
)]
fn boolean_infix_expression_cases(code: &str, snapshot_name: &str) {}

#[apply(boolean_infix_expression_cases)]
fn test_boolean_infix_expression_lexical_analysis(code: &str, snapshot_name: &str) {
    assert_lexical_analysis!(code, snapshot_name);
}

#[apply(boolean_infix_expression_cases)]
fn test_boolean_infix_expression_syntax_analysis(code: &str, snapshot_name: &str) {
    assert_syntax_analysis!(code, snapshot_name);
}

#[apply(boolean_infix_expression_cases)]
fn test_boolean_infix_expression_evaluator(code: &str, snapshot_name: &str) {
    assert_evaluator!(code, snapshot_name);
}

#[apply(boolean_infix_expression_cases)]
fn test_boolean_infix_expression_environment(code: &str, snapshot_name: &str) {
    assert_environment!(code, snapshot_name);
}
