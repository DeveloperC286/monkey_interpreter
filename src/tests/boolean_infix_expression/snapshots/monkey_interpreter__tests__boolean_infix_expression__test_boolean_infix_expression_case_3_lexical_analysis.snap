---
source: src/tests/boolean_infix_expression/mod.rs
expression: tokens
---
[
    True,
    Equals,
    False,
]
